assets/src/blocks/components/product-variant/index.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * Product variant select component.
+ */
+
+import { getProductVariations } from '../utilities';
+
+import PropTypes from 'prop-types';
+
+const { __ } = wp.i18n;
+const { useState, useEffect } = wp.element;
+const {
+	SelectControl,
+} = wp.components;
+
+const FastWCProductVariant = ( {
+	onChange,
+	product,
+	variant,
+} ) => {
+	const selectVariationOption = {
+		label: __( 'Select a variation...' ),
+		value: '',
+	};
+	const noVariantsOption = {
+		label: __( 'No variations available' ),
+		value: '',
+	};
+	const defaultOptions = [
+		{
+			label: __( 'Loading variations...' ),
+			value: '',
+		},
+	];
+
+	const [options, setOptions] = useState( defaultOptions );
+
+	useEffect(
+		() => {
+			getProductVariations( product )
+				.then( ( list ) => {
+					let options = [];
+					let variant_ids = [];
+
+					if ( list.length ) {
+						options = list.map( ( variant ) => ( {
+							label: variant.variation,
+							value: variant.id,
+						} ) );
+						variant_ids = list.map( ( variant ) => variant.id );
+						options.unshift( selectVariationOption );
+					} else {
+						options = [ noVariantsOption ];
+					}
+
+					if ( ! variant || ! variant_ids.includes( variant ) ) {
+						onChange( '' );
+					}
+
+					setOptions( options );
+				} );
+		},
+		[ product ]
+	);
+
+	const onSelect = ( variant ) => {
```

Comment:
same nit as other file
```suggestion
	const handleSelect = ( variant ) => {
```